src/buf/uninit_slice.rs

```diff
@@ -22,42 +22,42 @@ use core::ops::{
 pub struct UninitSlice([MaybeUninit<u8>]);
 
 impl UninitSlice {
-    /// Creates a `&mut UninitSlice` wrapping slice of uninitialised memory.
+    /// Creates a `&mut UninitSlice` wrapping a slice of initialised memory.
     ///
     /// # Examples
     ///
     /// ```
     /// use bytes::buf::UninitSlice;
-    /// use core::mem::MaybeUninit;
-    ///
-    /// let mut buffer = [MaybeUninit::uninit(); 64];
-    /// let slice = UninitSlice::from_uninit_slice(&mut buffer[..]);
     ///
-    /// let mut vec = Vec::with_capacity(1024);
-    /// let spare: &mut UninitSlice = vec.spare_capacity_mut().into();
+    /// let mut buffer = [0u8; 64];
+    /// let slice = UninitSlice::new(&mut buffer[..]);
     /// ```
     #[inline]
-    pub fn from_uninit_slice(slice: &mut [MaybeUninit<u8>]) -> &mut UninitSlice {
-        unsafe { &mut *(slice as *mut [MaybeUninit<u8>] as *mut UninitSlice) }
-    }
-
-    fn from_uninit_slice_ref(slice: &[MaybeUninit<u8>]) -> &UninitSlice {
-        unsafe { &*(slice as *const [MaybeUninit<u8>] as *const UninitSlice) }
+    pub fn new(slice: &mut [u8]) -> &mut UninitSlice {
+        unsafe { &mut *(slice as *mut [u8] as *mut [MaybeUninit<u8>] as *mut UninitSlice) }
     }
 
```

Comment:
I guess the only reason this isn't a breaking change is that we haven't yet released the `from_uninit_slice` method.

Reply:
Right.  v1.4.0 had only `from_raw_parts_mut` public constructor and
it’s not changed.  `from_slice` is changed but it’s `pub(crate)` so it
doesn’t affect the API.